exec/java-exec/src/main/java/org/apache/drill/exec/server/rest/profile/ProfileWrapper.java

```diff
@@ -63,7 +63,7 @@
 
   public ProfileWrapper(final QueryProfile profile, DrillConfig drillConfig) {
     this.profile = profile;
-    this.id = QueryIdHelper.getQueryId(profile.getId());
+    this.id = profile.getId().hasText() ? profile.getId().getText() : QueryIdHelper.getQueryId(profile.getId());
```

Comment:
both of them should return the same value. isn't it ?

Reply:
Yes. But if the profile is of an older format, ``hasText()` will ensure that the query ID text is decoded from the ID components: part1 & part2.

Reply:
why not decode always instead of doing the check ? It is not a big deal either way.
LGTM +1.

Reply:
The JIRA's goal is to embed the query ID text within the profile. Since it is within the profile, it would be more efficient to use that value versus decoding repeatedly. Hence, it is an optional part of the QueryID (see proto file), so that we can fall back to the decoding approach.